FirebaseAuth/Sources/User/FIRUser.m

```diff
@@ -282,28 +282,34 @@ + (void)retrieveUserWithAuth:(FIRAuth *)auth
   user.auth = auth;
   user.tenantID = auth.tenantID;
   user.requestConfiguration = auth.requestConfiguration;
-  [user internalGetTokenWithCallback:^(NSString *_Nullable accessToken, NSError *_Nullable error) {
-    if (error) {
-      callback(nil, error);
-      return;
-    }
-    FIRGetAccountInfoRequest *getAccountInfoRequest =
-        [[FIRGetAccountInfoRequest alloc] initWithAccessToken:accessToken
-                                         requestConfiguration:auth.requestConfiguration];
-    [FIRAuthBackend
-        getAccountInfo:getAccountInfoRequest
-              callback:^(FIRGetAccountInfoResponse *_Nullable response, NSError *_Nullable error) {
-                if (error) {
-                  // No need to sign out user here for errors because the user hasn't been signed in
-                  // yet.
-                  callback(nil, error);
-                  return;
-                }
-                user.anonymous = anonymous;
-                [user updateWithGetAccountInfoResponse:response];
-                callback(user, nil);
-              }];
-  }];
+  if (refreshToken.length) {
```

Comment:
nit: Maybe to avoid indenting this entire section, do something like:

```
if (!refreshToken.length) {
  [user updateWithIdToken:accessToken];
  callback(user, nil);
  return;
}

[user internalGetTokenWithCallback:^( ... ) {
  ...

}];

```